### ISSUE TITLE
[Question] Different MAPE between best run and final predictions?

### ISSUE THREAD

**Issue by gsamaras** [Nixtla/neuralforecast#267]
I am running [nhits__example.ipynb](https://colab.research.google.com/drive/1WjBbQzaivQhOldGolzymOtLmo6QX4Ieg#scrollTo=z9YUtuhNdsBA) as it is, with just including MAPE, e.g. here:

```
from neuralforecast.losses.numpy import mae, mape

model.fit(Y_df=Y_df, 
         ...
          loss_function_val=nf.losses.numpy.mae, # MSE, QL, MQL, sMAPE ...
          loss_functions_test={'mape':nf.losses.numpy.mape,
                                               'mae':nf.losses.numpy.mae})
```
and:
```
print('MAE: ', mae(y_hat, y_true))
print('MAPE: ', mape(y_hat, y_true))
```
gives:
```
MAE:  0.29348946
MAPE:  293.19941997528076
```
but:
```
trials.trials[best_run]['result']['test_losses']
```
gives:
```
{'mae': 0.29348946, 'mape': 113.8498306274414}
```
which is significantly better than the one computed manually by me with the predictions. I thought the predictions would be made with the best run during fitting the model. Isn't that the case? I mean why the two MAPEs are so different?

I also observed the same behavior with my data.

**Comment by kdgutier** [comment]
There are several things that could be happening:
1. MAPE is not a symmetric function, MAPE(y, y_hat) != MAPE(y_hat, y)
2. MAPE is known to degenerate when the target variable is close to 0, are no adding any numeric protection? The MAE suggests that you are using normalized data.
